src/main/java/org/ms/zk/plugin/matcher/HostMatcher.java

```diff
@@ -0,0 +1,15 @@
+package org.ms.zk.plugin.matcher;
+
+import com.unboundid.ldap.sdk.LDAPException;
+
+import java.io.IOException;
+
+public class HostMatcher extends Matcher {
+
+  @Override
+  public boolean matches(String id, String aclExpr) throws IOException, LDAPException {
+    return aclExpr.startsWith("host/")
+      && id.contains("@")
+      && aclExpr.equals(id.split("@", 2)[0]);
```

Comment:
REALM matching is there in src/main/java/org/ms/zk/plugin/matcher/Matcher.java and gets invoked always. The realm value is picked up from the config file